src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts

```diff
@@ -601,4 +601,72 @@ describe('TableComponent', () => {
       expect(component.useCustomClass('https://secure.it')).toBe('btn secure');
     });
   });
+
+  describe('test expand and collapse feature', () => {
+    beforeEach(() => {
+      component.identifier = 'a';
+      component.data = createFakeData(10);
+      component.expanded = _.clone(component.data[1]);
+    });
+
+    it('should update expaneded on refresh - "always"', () => {
+      spyOn(component.setExpandedRow, 'emit');
+      component.updateExpandedOnRefresh = 'always';
+
+      component.data[1].b = 'test';
+      component.updateExpanded();
+      expect(component.expanded.b).toBe('test');
+      expect(component.setExpandedRow.emit).toHaveBeenCalled();
+    });
+
+    it('should update expaneded on refresh - "onChange" with changes', () => {
+      spyOn(component.setExpandedRow, 'emit');
+      component.updateExpandedOnRefresh = 'onChange';
+
+      component.data[1].b = 'test';
+      component.updateExpanded();
+      expect(component.expanded.b).toBe('test');
+      expect(component.setExpandedRow.emit).toHaveBeenCalled();
+    });
+
+    it('should update expaneded on refresh - "onChange" without changes', () => {
+      spyOn(component.setExpandedRow, 'emit');
+      component.updateExpandedOnRefresh = 'onChange';
+
+      component.updateExpanded();
+      expect(component.expanded.b).toBe(10);
+      expect(component.setExpandedRow.emit).not.toHaveBeenCalled();
+    });
+
+    it('should update expaneded on refresh - "never"', () => {
+      spyOn(component.setExpandedRow, 'emit');
+      component.updateExpandedOnRefresh = 'never';
+
+      component.data[1].b = 'test';
+      component.updateExpanded();
+      expect(component.expanded.b).toBe(10);
+      expect(component.setExpandedRow.emit).not.toHaveBeenCalled();
+    });
+
+    it('should open the table details and close other expanded rows', () => {
+      spyOn(component.setExpandedRow, 'emit');
+      component.table = {
+        rowDetail: { collapseAllRows: jest.fn(), toggleExpandRow: jest.fn() }
+      } as any;
+      component.toggleExpandRow(component.expanded, false);
+      expect(component.expanded).toBe(component.expanded);
```

Comment:
This could be removed but I spotted what you wanted to test here so I suggest the following:
```suggestion
      expect(component.expanded).toEqual({ a: 1, b: 10, c: true });
```